packages/types/src/codec/Base.ts

```diff
@@ -2,14 +2,73 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
+import { AnyJson, Codec } from '../types';
+
 /**
  * @name Base
  * @description A type extends the Base class, when it holds a value
  */
-export default class Base<T> {
+export default abstract class Base<T extends Codec> implements Codec {
```

Comment:
To make extension less noisy, the base defaults have been included into Base. Anything that extends it have been adjusted to remove duplication. (This happened since I really didn't want to add more boilerplate to the new classes, so rather cleaned up all)